src/index.js

```diff
@@ -134,12 +134,14 @@ export function asOptional(cleaner, fallback) {
 /**
  * Makes a cleaner that accepts either of the given types.
  */
-export function asEither(a, b) {
+export function asEither(...shape) {
   return function asEither(raw) {
-    try {
-      return a(raw)
-    } catch (e) {
-      return b(raw)
+    for (let i = 0; i < shape.length; i++) {
+      try {
+        return shape[i](raw)
+      } catch (error) {
+        if (i + 1 >= shape.length) throw error
+      }
```

Comment:
I feel it would be worth it to aggregate errors here. We could use [AggregateError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError) or roll our own TypeError that includes a useful message like: `"TypeError: Expected a or b or c at .unionProp"`. This might require a bit of re-thinking the error handling, but this was one of the outstanding things I thought could be fixed with asEither error messages when first using them.

Reply:
My suggestion can be done in a separate PR.